src/Compiler/Service/FSharpWorkspaceQuery.fsi

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+module FSharp.Compiler.CodeAnalysis.Workspace.FSharpWorkspaceQuery
+
+open System
+open System.Threading.Tasks
+
+open Internal.Utilities.DependencyGraph
+
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.CodeAnalysis.ProjectSnapshot
+open FSharp.Compiler.CodeAnalysis.Workspace.FSharpWorkspaceState.WorkspaceGraphTypes
+open FSharp.Compiler.Diagnostics
+open FSharp.Compiler.EditorServices
+open FSharp.Compiler.Text
+
+[<Experimental("This FCS API is experimental and subject to change.")>]
+type FSharpDiagnosticReport =
+
+    internal new: diagnostics: FSharpDiagnostic array * resultId: int -> FSharpDiagnosticReport
+
+    member Diagnostics: FSharpDiagnostic array with get
+
+    /// The result ID of the diagnostics. This needs to be unique for each version of the document in order to be able to clear old diagnostics.
+    member ResultId: string with get
```

Comment:
Does this have to be out of the box serializable?
If not, I think we could design+mark the types and not use plain primitives.

Reply:
This is a bit of a helper for LSP. In the protocol it's a string and probably it won't be used/useful for anything else. Maybe wait how all this evolves and then see if we want some more types around it?